### ISSUE TITLE
Collection#get should have a variant that doesn't throw when records don't exist

### ISSUE THREAD

**Issue by glasserc** [Kinto/kinto.js#443]
It seems that at present, `Collection#get`, when given the ID of a record that doesn't exist, just throws a specific `Error`. This is hard to test for. As a result, if you want to check to see if a record is in the database, you have to call `get()` with a `catch` that examines the string message to see if it contains "not found". It would be nice if there was a clearer way to try to `get()` in a safe (non-throwing) way.

**Comment by n1k0** [comment]
> This is hard to test for.

You could alternatively a little reusable helper which would be much easier to use in your project:

```js
function getRecord(id) {
  return coll.get(id).catch(err => {
    if (/not found/.test(err.message)) {
      return;
    }
    throw err;
  });
}

getRecord("non-existent")
  .then(res => [=> undefined])
```

**Comment by glasserc** [comment]
Yes, this is actually what the existing code for https://bugzilla.mozilla.org/show_bug.cgi?id=1253740 does. That's what I meant when I said that it was "hard to test for". I think running a regular expression to check for a certain kind of exception is incredibly ugly. At the very least, I'd want a separate kind of exception, although with inheritance being the way it is in JS, that might not be advisable.

**Comment by n1k0** [comment]
Yeah we're in js userland so typed exceptions is no easy business. It's possible to *subclass* `Error` but in my own experience itbrings its share of nightmares. Though if the regexp thing is your only gripe, would exposing a `code` property to the error help you feeling better? eg. `if (err.type === NOT_FOUND))` or eq.

Yeah that would basically be monkey patching the error object instance, so you might find this incredibly ugly too. So I don't know, maybe I'm too much used to this crap.

**Comment by glasserc** [comment]
This got implemented as `getRaw()` in #455.
